tests/Integration/Classes/CartRuleTest.php

```diff
@@ -0,0 +1,200 @@
+
+    /**
+     * @var Configuration
+     */
+    public $configuration;
+
+    public static function setUpBeforeClass(): void
+    {
+        DatabaseDump::restoreDb();
+    }
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->dummyCustomer = $this->createDummyCustomer();
+        $this->configuration = new Configuration();
+    }
+
+    public function testGetCartRulesForCustomer(): void
+    {
+        $this->dummyCartRule = $this->createDummyCartRule(true, (int) $this->dummyCustomer->id);
+
+        $customerCartRules = CartRule::getCustomerCartRules(
+            $this->configuration->get('PS_LANG_DEFAULT'),
+            $this->dummyCustomer->id
+        );
+
+        $this->assertEquals(1, count($customerCartRules));
+    }
+
+    public function testGetAllCartRulesForCustomerEvenDisabled(): void
+    {
+        $this->dummyCartRule = $this->createDummyCartRule(false, (int) $this->dummyCustomer->id);
+
+        $customerCartRules = CartRule::getCustomerCartRules(
+            $this->configuration->get('PS_LANG_DEFAULT'),
+            $this->dummyCustomer->id,
+            true
+        );
+
+        // We assert 0 because 'active' flag does not work in "CartRule::getCustomerCartRules"
+        // because of CartRule::isFeatureActive and one additional check
+        // which doesn't not work if we have only 1 customer rule or all of them are disabled
+        // see https://github.com/PrestaShop/PrestaShop/issues/21556 for more details
+        $this->assertEquals(0, count($customerCartRules));
+    }
+
+    public function testGetAllCartRulesForCustomerWithDedicatedMethod(): void
+    {
+        $this->dummyCartRule = $this->createDummyCartRule(true, (int) $this->dummyCustomer->id);
+
+        $customerCartRules = CartRule::getAllCustomerCartRules(
+            (int) $this->dummyCustomer->id
+        );
+
+        $this->assertEquals(1, count($customerCartRules));
+    }
+
+    public function testGetAllCartRulesForCustomerWithDedicatedMethodEvenDisabled(): void
+    {
+        $this->dummyCartRule = $this->createDummyCartRule(false, (int) $this->dummyCustomer->id);
+
+        $customerCartRules = CartRule::getAllCustomerCartRules(
+            (int) $this->dummyCustomer->id
+        );
+
+        $this->assertEquals(1, count($customerCartRules));
+    }
+
+    public function testGetAllCartRulesForCustomerWithDedicatedMethodBothEnabledAndDisabled()
+    {
+        $this->createDummyCartRule(false, (int) $this->dummyCustomer->id);
+        $this->createDummyCartRule(true, (int) $this->dummyCustomer->id);
+        $this->createDummyCartRule(true, (int) $this->dummyCustomer->id);
+
+        $customerCartRules = CartRule::getAllCustomerCartRules(
+            (int) $this->dummyCustomer->id
+        );
+
+        $this->assertEquals(3, count($customerCartRules));
+    }
+
+    public function testGetAllCartRulesForCustomerWithDedicatedMethodBothEnabledAndDisabledWithOtherCustomerCartRulesAvailable()
```

Comment:
```suggestion
    public function testGetAllCartRulesForCustomerWithDedicatedMethodBothEnabledAndDisabledWithOtherCustomerCartRulesAvailable(): void
```
Magic Name... Do you like Java ?

Reply:
😅 I like to read code you know, 

normally I'd go with:

`test_if_this_is_working_or_not` instead of camel case but hey! we still can read it in camelCase :D